backend/rds.ts

```diff
@@ -89,12 +92,39 @@ export async function ensureDatabase() {
 
 async function createDatabase() {
   await executeStatementInDatabase(null, "CREATE DATABASE " + dbName);
-  await executeStatement(
-    "CREATE TABLE Client (Id VARCHAR(255) PRIMARY KEY NOT NULL, LastMutationID BIGINT NOT NULL)"
-  );
-  await executeStatement(
-    "CREATE TABLE Shape (Id VARCHAR(255) PRIMARY KEY NOT NULL, Content TEXT)"
-  );
+
+  await executeStatement(`CREATE TABLE Cookie (
+    Version BIGINT NOT NULL)`);
+  await executeStatement(`CREATE TABLE Client (
+    Id VARCHAR(255) PRIMARY KEY NOT NULL,
+    LastMutationID BIGINT NOT NULL)`);
+  await executeStatement(`CREATE TABLE Shape (
+    Id VARCHAR(255) PRIMARY KEY NOT NULL,
+    Content TEXT NOT NULL,
+    Version BIGINT NOT NULL)`);
+
+  await executeStatement(`INSERT INTO Cookie (Version) VALUES (0)`);
+
+  // To calculate which rows have changed for return in the pull endpoint, we
+  // use a very simply global version number. This is easy to understand but
+  // does serialize writes.
+  //
+  // There are many different strategies for calculating changed rows and the
+  // details are very dependent on what you are building. Contact us if you'd
+  // like help: https://replicache.dev/#contact.
+  await executeStatement(`CREATE PROCEDURE NextVersion (OUT result BIGINT)
+    BEGIN
+      UPDATE Cookie SET Version = Version + 1;
+      SELECT Version INTO result FROM Cookie;
+    END`);
+
+  await executeStatement(`CREATE PROCEDURE PutShape (IN pId VARCHAR(255), IN pContent TEXT)
```

Comment:
the "p" in "pid" and "pcontent" confused me for a sec. i think it means "put"? maybe more clear to call it "s"id etc for shape? (or maybe this is a naming convention thing -- p for param?)